### PR TITLE
Configure GPS

### PR DESCRIPTION
**Pull request by wiktorwieclaw**
* Configure output rates of NMEA messages using ublox's proprietary PBUX NMEA messages, eg. `$PUBX,40,GGA,0,1,0,0,0,0*5B`. GGA is setup so that it is sent on each epoch and all the other messages are disabled. Refer to [ublox specification](https://content.u-blox.com/sites/default/files/products/documents/u-blox7-V14_ReceiverDescriptionProtocolSpec_%28GPS.G7-SW-12001%29_Public.pdf) for details, specifically sections 25.3 and 29.
* Add generic MAX_NMEA_LEN argument to Gps. It turns out that some devices implement messages that do not adhere to the standard 82 characters limit.
* Bump `stm32f4xx_hal` to 0.16
* Remove Default from Temperature. It really shouldn't impl Default as it is not obvious what the default value should be, considering that the type handles both, Celsius and Kelvins.

### CODE REVIEW

**Comment by wiktorwieclaw** [comment]
Closes #52

File: crates/cansat-stm32f4/src/startup.rs [hunk 1]
```diff
@@ -149,11 +146,31 @@ fn init_bme280(i2c: I2c1Proxy, delay: &mut Delay) -> Result<Bme280, Bme280Error>
...
+    let set_gga_output_rate = b"$PUBX,40,GGA,0,1,0,0,0,0*5B\r\n";
+    gps.send(set_gga_output_rate)?;
+
+    let set_vtg_output_rate = b"$PUBX,40,VTG,0,0,0,0,0,0*5E\r\n";
+    gps.send(set_vtg_output_rate)?;
+
+    let set_rmc_output_rate = b"$PUBX,40,RMC,0,0,0,0,0,0*47\r\n";
+    gps.send(set_rmc_output_rate)?;
+
+    Ok(gps)
```
Comment (PatrykPaluch) [inline crates/cansat-stm32f4/src/startup.rs:173]: I haven't find any docs for this, only one [arduino lib](https://github.com/Velleman/VMA430_GPS_Module/tree/master) that uses different values to initialize module. Can you provide link to docs or explain those values? You already tested this, so it probably works, but im curious :P

File: crates/cansat-stm32f4/src/startup.rs [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
+    spi::{self, Spi2},
     timer::{monotonic::MonoTimerUs, DelayUs},
 };
 use tap::prelude::*;
 
 pub type Monotonic = MonoTimerUs<pac::TIM2>;
 pub type Delay = DelayUs<pac::TIM3>;
 pub type Led = gpio::PC13<gpio::Output>;
-pub type Gps = cansat_gps::Gps<Serial1>;
+pub type Gps = cansat_gps::Gps<Serial1, 256>;
```
[the remark this answers is not in the document]
Comment (wiktorwieclaw) [reply crates/cansat-stm32f4/src/startup.rs:18]: actually I wanted to remove this constant. I don't want to assume a max size for the library user.

>  It turns out that some devices implement messages that do not adhere to the standard 82 characters limit.
can't really predict what bonkers messages are implemented out there

[the remark this answers is not in the document]
Comment (wiktorwieclaw) [reply crates/cansat-stm32f4/src/startup.rs:18]: good catch either way :P

[the remark this answers is not in the document]
Comment (wiktorwieclaw) [reply crates/cansat-stm32f4/src/startup.rs:18]: i guess ill just rename it to STANDARD_MAX_NMEA_LEN

File: crates/cansat-core/src/measurements.rs [hunk 3]
```diff
@@ -1,91 +1,93 @@
...
+    v.map(orientation_to_str).serialize(serializer)
+}
+
+impl defmt::Format for Measurements {
+    fn format(&self, fmt: defmt::Formatter) {
+        defmt::write!(
+            fmt,
+            "temp: {} °C, pres: {} hPa, alt: {} m, accel: {}, orient: {}, nmea: {=[u8]:a}",
+            self.temperature
+                .unwrap_or(Temperature::from_celsius(0.0))
```
[the remark this answers is not in the document]
Comment (wiktorwieclaw) [reply crates/cansat-core/src/measurements.rs:82]: Perhaps I should just implement a custom wrapper that implements defmt::Format that will format to nothing if None. I'll make a commit in a sec

File: crates/cansat-stm32f4/src/startup.rs (hunk 1 above)
Comment (wiktorwieclaw) [reply crates/cansat-stm32f4/src/startup.rs:173]: [25.3](https://content.u-blox.com/sites/default/files/products/documents/u-blox7-V14_ReceiverDescriptionProtocolSpec_%28GPS.G7-SW-12001%29_Public.pdf)

File: crates/cansat-stm32f4/src/startup.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (wiktorwieclaw) [reply crates/cansat-stm32f4/src/startup.rs:18]: Actually.... I'm not even sure whether Im even correct on this one. How would you interpret this [note](https://receiverhelp.trimble.com/alloy-gnss/en-us/NMEA-0183messages_GGA.html#:~:text=NOTE%20%E2%80%93%20The%20data%20string%20exceeds%20the%20NMEA%20standard%20length)?

**Pull request merged**